### PR TITLE
Extract prisma to it's own package

### PR DESCRIPTION
**Pull request by zomars**
## What does this PR do?

refs #1815 #1581 

- Extract all prisma logic into a separate package so we can reuse un website and admin console.
- Extracted some lib and types used in the seeder as well.
- Fixed some missing linters
- Updates docs to reflect these new changes
- Mostly sort imports because of plugin changes

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How should this be tested?

- [ ] `yarn build`
- [ ] `yarn dx`

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code and corrected any misspellings
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: packages/lib/availability.ts
Comment (zomars) [inline packages/lib/availability.ts:1]: Moved here since the seeder also uses this logic and to avoid circular dependencies.

**Review by zomars — COMMENTED** [review]
Ready for review

**Comment by zomars** [comment]
blocked by #1826

**Review by pumfleet — APPROVED** [review]
Code looks good as far as I can see, but let's get another review from someone more intelligent than me before merging :)

**Pull request merged**
